src/modules/task-show.js

```diff
@@ -0,0 +1,130 @@
+let toDotasksupdate = '';
+
+/* || FUNCTION TO DISPLAY THE TASKS */
+export const showToDoTasks = () => {
+  if (Object.keys(localStorage).length > 0) {
+    toDoDivTask.style.display = 'inline-block';
+  } else {
+    toDoDivTask.style.display = 'none';
+  }
+  /* || Clear the tasks */
+  toDoDivTask.innerHTML = '';
+  /* || Fetch All The Keys in local storage */
+  let toDoTasks = Object.keys(localStorage);
+  toDoTasks = toDoTasks.sort();
+
+  toDoTasks.forEach((key) => {
+    /* || Get all values */
+    const toDoValue = localStorage.getItem(key);
+
+    const toDoInnerDiv = document.createElement('p');
+    toDoInnerDiv.classList.add('p__cont__task');
+    toDoInnerDiv.setAttribute('id', key);
+    toDoInnerDiv.innerHTML = `<p id="todo__name">${key.split('_')[1]}</p>`;
+    /* || localstorage would store boolean as string so we parse it to boolean back */
+    const buttonModify = document.createElement('button');
+    buttonModify.classList.add('edit__btn');
+    buttonModify.innerHTML = '<i class="fa-solid fa-pen-to-square"></i>';
+    if (!JSON.parse(toDoValue)) {
+      buttonModify.style.visibility = 'visible';
+    } else {
+      buttonModify.style.visibility = 'hidden';
+      toDoInnerDiv.classList.add('completedTask');
+    }
+    toDoInnerDiv.appendChild(buttonModify);
+    toDoInnerDiv.innerHTML += '<button class="del__btn"><i class="fa-solid fa-trash"></i></button>';
+    toDoDivTask.appendChild(toDoInnerDiv);
+  });
+  /* || tasks completed */
+  toDoTasks = document.querySelectorAll('.p__cont__task');
+  toDoTasks.forEach((element) => {
+    element.onclick = () => {
+      /*  || local storage update */
+      if (element.classList.contains('completedTask')) {
+        storageUpdate(element.id.split('_')[0], element.innerText, false);
+      } else {
+        storageUpdate(element.id.split('_')[0], element.innerText, true);
+      }
+    };
+  });
+  /* || Edit Tasks */
+  modifyToDoTasks = document.getElementsByClassName('edit__btn');
+  Array.from(modifyToDoTasks).forEach((element) => {
+    element.addEventListener('click', (e) => {
+      e.stopPropagation();
+      /* || disable other edit buttons when one task is being edited */
+      buttonsDeactive(true);
+      /* || update input value and remove div */
+      const parentEl = element.parentElement;
+      inputNewTasks.value = parentEl.querySelector('#todo__name').innerText;
+      /* || set updateNote to the task that is being edited */
+      toDotasksupdate = parentEl.id;
+      /*  || remove task */
+      parentEl.remove();
+    });
+  });
+  /*   || Delete Tasks */
+  clearToDoTasks = document.getElementsByClassName('del__btn');
+  Array.from(clearToDoTasks).forEach((element) => {
+    element.addEventListener('click', (e) => {
+      e.stopPropagation();
+      /* || Delete from local storage and remove div */
+      const parentEl = element.parentElement;
+      deleteTask(parentEl.id);
+      parentEl.remove();
+      counter -= 1;
+    });
+  });
```

Comment:
Hello, @mrkamin your project is looking and working great 😄!
As requested on the project requirements:
![image](https://user-images.githubusercontent.com/42799579/202858933-e8928178-406d-4bdd-aebb-652c52194cb5.png)

Your tasks should have their indexes updated after being deleted. Currently, the lack of this feature is causing a bug 😉:
If I add tasks with the same name:
![image](https://user-images.githubusercontent.com/42799579/202858979-db30fdd4-4f60-4909-a1ee-b46630a2573c.png)

And delete one of them:
![image](https://user-images.githubusercontent.com/42799579/202858993-e84eecac-6e74-4204-b039-da1a73369584.png)
Please notice that the indexes weren't updated.

Then, if I try to add a new one with the same name, it doesn't work:
![image](https://user-images.githubusercontent.com/42799579/202859007-79ab8ed4-e8d7-4ec6-b6a7-99bef33e8ae0.png)

I know this sounds like something that may never happen with your actual code, but as programmers, we need to make sure our code works as intended for every possible scenario 😄. Please make sure to update your indexes after removing a task 😉. Cheers!

Reply:
@Gopxfs,
thanks for your good and valuable feedback.
I will take in to account your comments.